internal/controller/apiextensions/composite/composed.go

```diff
@@ -344,8 +344,8 @@ func (r *APIDryRunRenderer) Render(ctx context.Context, cp resource.Composite, c
 	cd.SetNamespace(namespace)
 
 	onlyPatches := []v1.PatchType{v1.PatchTypeFromCompositeFieldPath, v1.PatchTypeCombineFromComposite}
-	for i, p := range t.Patches {
-		if err := p.Apply(cp, cd, onlyPatches...); err != nil {
+	for i := 0; i < len(t.Patches); i++ {
+		if err := t.Patches[i].Apply(cp, cd, onlyPatches...); err != nil {
```

Comment:
I was using `Patch.Filtered` as a cache to store the filtering result computed during a call to `Patch.Apply`. If I remember correctly, `p.Apply` was storing the filtering result in the copied loop variable, not in the actual `Patch` object. So, I had switched to indexing `t.Patches` to have `Filtered` set on the actual object. I have reverted this because I'm no longer using a `Patch.Filtered` to cache filtering computation.

Reply:
Ooops, it turns out that we were also caching the computation of the default value of `Patch.ToFieldPath` if it's not set. I did not duplicate it elsewhere, but reverted back to indexing of `t.Patches`.